usr/src/uts/armv8/io/arm_gtmr.c

```diff
@@ -0,0 +1,591 @@
+}
+
+static uint_t
+cbe_softclock(caddr_t arg1 __unused, caddr_t arg2 __unused)
+{
+	cyclic_softint(CPU, CY_LOCK_LEVEL);
+	return (1);
+}
+
+static uint_t
+cbe_low_level(caddr_t arg1 __unused, caddr_t arg2 __unused)
+{
+	cyclic_softint(CPU, CY_LOW_LEVEL);
+	return (1);
+}
+
+/*
+ * Driver routines.
+ */
+
+static int
+arm_gtmr_attach(dev_info_t *dip, ddi_attach_cmd_t cmd)
+{
+	int ret;
+	int itypes;
+	int icount;
+	int iactual;
+	int iwanted;
+	uint_t intr_pri;
+	uint_t inum;
+	ddi_intr_handle_t *itable;
+	int def_prios[EXPECTED_NINTRS] = {
+		CBE_HIGH_PIL,
+		CBE_HIGH_PIL,
+		CBE_HIGH_PIL,
+		CBE_HIGH_PIL
+	};
+	uint_t def_nprios = EXPECTED_NINTRS;
+
+	cyc_backend_t cbe = {
+		.cyb_configure		= cbe_configure,
+		.cyb_unconfigure	= cbe_unconfigure,
+		.cyb_enable		= cbe_enable,
+		.cyb_disable		= cbe_disable,
+		.cyb_reprogram		= cbe_reprogram,
+		.cyb_softint		= cbe_softint,
+		.cyb_set_level		= cbe_set_level,
+		.cyb_restore_level	= cbe_restore_level,
+		.cyb_xcall		= cbe_xcall,
+		.cyb_suspend		= cbe_suspend,
+		.cyb_resume		= cbe_resume
+	};
+
+	cyc_handler_t hdlr;
+	cyc_time_t when;
+
+	switch (cmd) {
+	case DDI_ATTACH:
+		break;
+	case DDI_RESUME:	/* fallthrough */
+	case DDI_PM_RESUME:
+		return (DDI_SUCCESS);
+	default:
+		return (DDI_FAILURE);
+	}
+	ASSERT3U(cmd, ==, DDI_ATTACH);
+	ASSERT3U(ddi_get_instance(dip), ==, 0);
+
+	arch_timer_disable();
+	arch_timer_mask_irq();
+	CPUSET_ZERO(cbe_enabled);
+
+	mutex_init(&cbe_xcall_lock, NULL, MUTEX_DEFAULT, NULL);
+
+	/*
+	 * Per the bindings document there are four interrupts defined for
+	 * the armv8 architected timer, these are (with their indices):
+	 * - 0: secure physical
+	 * - 1: non-secure physical
+	 * - 2: virtual
+	 * - 3: hypervisor
+	 *
+	 * If booted at EL1 we use the virtual timer (index 2), otherwise we
+	 * use the non-secure physical timer (index 1). We only ever use one
+	 * of the offered interrupts.
+	 *
+	 * While it is possible to defined a driver.conf(5) for this driver,
+	 * it is highly discouraged, as the only proprty that may make sense
+	 * is `interrupt-priorities'. If present, `interrupt-priorities' must
+	 * contain an array of four integers and each integer must have the
+	 * value 14 (corresponding to CBE_HIGH_PIL), or the cyclic backend
+	 * will not behave correctly.
+	 */
+
+	if (!ddi_prop_exists(DDI_DEV_T_ANY, dip,
+	    DDI_PROP_DONTPASS, OBP_INTERRUPT_PRIORITIES)) {
+		if ((ret = ndi_prop_update_int_array(DDI_DEV_T_NONE, dip,
+		    OBP_INTERRUPT_PRIORITIES,
+		    def_prios, def_nprios)) != DDI_PROP_SUCCESS)
+			dev_err(dip, CE_PANIC,
+			    "Failed to set default priorities");
+	}
+
+	iwanted = 1;
+
+	inum = ARM_GTMR_INUM_PHYS_NS;
+	if (CPU->cpu_m.mcpu_boot_el == 1)
+		inum = ARM_GTMR_INUM_VIRTUAL;
+
+	/*
+	 * We can only use fixed interrupts
+	 */
+	ret = ddi_intr_get_supported_types(dip, &itypes);
+	if ((ret != DDI_SUCCESS) || (!(itypes & DDI_INTR_TYPE_FIXED)))
+		dev_err(dip, CE_PANIC, "Fixed type interrupt is not supported");
```

Comment:
You have your error message backwards :)

Reply:
Do I? If it fails _or_ if `FIXED` is not in types, then panic? Or do you mean the text itself?